packages/driver/src/cypress/command.ts

```diff
@@ -53,21 +45,17 @@ export class $Command {
         log.snapshot()
       }
 
+      if (!log.get('snapshots')) {
```

Comment:
nit: `if (log.get('next') || !log.get('snapshots'))`

If I'm understanding correctly, we are requerying each time a chained command executed? So 
```
cy.get().should('...').should('...').click()

```

Reply:
Yeah, looks like it did. I has started to ask about some of the details but then started to double-check myself and I'm wondering if my original thoughts are valid. It looks like certain command are always re-executing the subject chain functions when the commands starts. Is that correct or is the requerying only happening on a retry if the subject is detached?

Reply:
That's correct; the subject is determined fresh at the start of every command or query. There are two reasons for this, principled and practical.

Consider the following test.

```
cy.$$('<button>Greetings.</button>').prependTo('body')

cy.get('button:first').as('btn')

setTimeout(() => {
  cy.$$('<button>Hello!</button>').prependTo('body')
}, 1000)

cy.wait(2000)

cy.get('@btn')
```

Should I get "Greetings" or "Hello!"? "Greetings" didn't detach from the DOM, so should we return it? But if it *had* detached - if we'd replaced it with a copy of itself before adding in "Hello", we'd get a different result!

So we always re-query for consistency's sake.

The more practical reason is so that I don't have to thread "the current subject" though the `command_queue` - we only need to track the subjectChain, not the subject *and* the subjectChain used to generate it.

Reply:
Is there impact on the overall test times by rerunning these every time?

Reply:
Good idea to measure, I last ran benchmarks a couple months ago. There does appear to be a cost.

```
it.only('is fast?', () => {
  cy.visit('/fixtures/dom.html')

  const now = Date.now()

  for (let i = 0; i < 10000; i++) {
    cy.get('button', { log: false }).first({ log: false }).focus({ log: false })
  }

  cy.then(() => {
    cy.log('end: ' + (Date.now() - now) + 'ms')
  })
})
```

This executes 20,000 queries and 10,000 commands. `{log: false}` is used, because otherwise the snapshotting / command log completely swamps any possible change; that's by far the slowest part of any command.

I ran this test three times against each branch. In `develop`, this takes an average of 10.037s, while in `issue-7306-addQuery-actionability`, it takes an average of 10.615s. 

About 5% slower for executing the queries - but keep in mind how extremely focused the test is. It's a 0.02ms (not a typo) overhead per command, I would not consider a problem.